### ISSUE TITLE
authentication error

### ISSUE THREAD

**Issue by c3p0vsr2d2** [twrecked/pyaarlo#153]
Seeing an authentication error for the past week or so ... is anyone else noticing similar issues?

**Comment by burner1024** [comment]
Doesn't work for me all.
Doesn't help that error messages are cryptic, even with debug settings on.

**Comment by twrecked** [comment]
@burner1024 not going to disagree, but what are you seeing? If you paste something in here I might be able to let you know what is going wrong.

**Comment by burner1024** [comment]
```
authentication failed
authentication failed
```
That's it.

**Comment by twrecked** [comment]
If you know your Arlo username and password this will be a cloudflare issue.

I guess I had been thinking about this recently. I just pushed a branch I've been working on - https://github.com/twrecked/pyaarlo/tree/better-errors

Can you try it? I started on it a few months ago and I'm not sure what state it is in and I won't get a change to try until later but it might help you.

**Comment by burner1024** [comment]
```
login failed: 403 - possible cloudflare issue
login failed: 403 - possible cloudflare issue
```

Strangely, _sometimes_ I get 
```
error in new response={'meta': {'code': 400, 'error': 9204, 'message': 'Browser is not trusted'}}
login failed: 2fa: no secondary choices available
login failed: 403 - possible cloudflare issue
```
When running exactly same code.

**Comment by twrecked** [comment]
Welcome to Cloudflare - where the errors are made up!

Try changing some of the values mentioned [here](https://github.com/twrecked/hass-aarlo?tab=readme-ov-file#how-we-work-around-this).

These are done from with arguments  to `PyArlo` not with `aarlo.yaml`, but I hope it makes sense.

**Comment by twrecked** [comment]
The 2fa thing might be working. Do you have 2fa set up for this account?

**Comment by burner1024** [comment]
Yes. I don't think there's an option to _not_ set it up, is there?

**Comment by twrecked** [comment]
I need how you are calling `PyArlo` and some debug. 

You can use [this link](https://github.com/twrecked/hass-aarlo?tab=readme-ov-file#encrypting-the-output) to encrypt the output.

**Comment by burner1024** [comment]
Thank you for your assistance @twrecked.
It appears that the IP address from which I was testing was simply blocked by Cloudflare. (I finally just visited https://ocapi-app.arlo.com/ in browser, and it didn't even offer a captcha). Maybe due to multiple login failures, maybe just Cloudflare algos going crazy.
Playing with ecdh_curve and user agent didn't help.

It works intermittently on another IP. I try some more debugging later.

Some suggestions:
- For bettter errors, if there's something like `data-translate="blocked_why_detail">This website is using a` in body, it's not a "possible" cloudflare issue, but a definite one.
- On the other hand, once there's `pyaarlo - WARNING - error in new response={'meta': {'code': 401, 'error': 9015, 'message': 'Password not correct'}}`, I think there's no point in retrying, can exit directly (right now it retries multiple times).

**Comment by twrecked** [comment]
Thanks for trying the new code. Cloudflare is always fun (!) to debug, I have 2 machines on the same network that need different setting to make them work...

Sometimes you just have to wait a few hours for Cloudflare to reset something. It might work later.

And suggestions noted, I'll get them into the error branch.
